3.5 fix out of application context

cornelinux opened the pull request:
Fix two problems in the tests - not the code itself.

cornelinux committed:
Fix out of application context problem

cornelinux committed:
fix usercache test

pep8speaks commented:
Hello @cornelinux! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`tests/test_lib_usercache.py`](https://github.com/privacyidea/privacyidea/blob/0a7b13115da47e89dcf032758945d8b065ee9e44/tests/test_lib_usercache.py):

> [Line 167:1](https://github.com/privacyidea/privacyidea/blob/0a7b13115da47e89dcf032758945d8b065ee9e44/tests/test_lib_usercache.py#L167): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

plettich commented on privacyidea/lib/resolvers/LDAPIdResolver.py:
```diff
@@ -99,9 +99,13 @@
+    pooling_loop_timeout = get_app_config_value("PI_LDAP_POOLING_LOOP_TIMEOUT", 10)
+    log.info("Setting system wide POOLING_LOOP_TIMEOUT to {0!s}.".format(pooling_loop_timeout))
+    ldap3.set_config_parameter("POOLING_LOOP_TIMEOUT", pooling_loop_timeout)
+except RuntimeError:
```
I'd rather not fix issues from tests in the code.
Maybe we can move this call to the ``__init__`` function of the LDAPIdResolver, if there is no performance impact. We should always have an application context there.

cornelinux replied:
I thought about putting it into `__init__`.
This also feels a bit wired, because init is called for each resolver, in each process. So the global LDAP3 setting is set several times.
I think the performance impact would not be...
...relevant.

plettich replied:
I also thought about putting it into the ``app.py`` as all the global stuff is set there (and the configuration freshly read). But then we would have to import ``ldap3`` there as well which seemed a little too much.
We could use the technique used in the queue implementation but this also seems like overkill: https://github.com/privacyidea/privacyidea/blob/907a3c15fd2b3233062331912216b533012c952e/privacyidea/lib/queue.py#L34

cornelinux replied:
Then I would prefer to put it into `__init__` leaving it in the LDAPResolver.py file.

Now, you?

cornelinux replied:
see 844fc170

cornelinux committed:
Avoid try-except for outside of application context

We can not read the config value in the class creation,
since this is also called in test_ldap_mock, and thus
runs outside of the application context.

So if we want to avoid having a try-except block we need
indeed put it into the __init__ method.

cornelinux committed:
Merge branch 'master' into 3.5_fix_out_of_application_context

cornelinux commented:
I will create a new PR
